src/mon/OSDMonitor.cc

```diff
@@ -2062,6 +2076,8 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   // health
   health_check_map_t next;
   tmp.check_health(cct, &next);
+  // OSD_FILESTORE
```

Comment:
Can we add the check for `__FreeBSD__` here?

Reply:
Makes sense. Also `__FreeBSD__` check should only be for `check_for_filestore_osds`or should we introduce in other places like for `get_filestore_osd_list` and while filling up filestore_osds ?

Reply:
From my point of view I'd like to remove the warning In HEALTH about depricated FileStore on FreeBSD as long as BlueStore is not available on FreeBSD. Otherwise the cluster will not be able to get into a HEALTHY state.
Things like warnings in the logs etc. would be fine. 

It is just that a FreeBSD cluster with FileStore OSDs is perfectly healty.

Come to think about it, I do not necessarilly agree that a cluster (not just FreeBSD) with FileStore OSDs in it would warrant a `HEALT_WARN` state. That state is a sort of allert that something in the cluster is requiring more that average attention. And I don't consider having FileStore OSDs in the cluster that high an alert.
So if it were up to me, I'd report it in the `ceph status` overview but still mark the cluster HEALTHY

Reply:
> From my point of view I'd like to remove the warning In HEALTH about depricated FileStore on FreeBSD as long as BlueStore is not available on FreeBSD. Otherwise the cluster will not be able to get into a HEALTHY state. Things like warnings in the logs etc. would be fine.
> 
> It is just that a FreeBSD cluster with FileStore OSDs is perfectly healty.
> 
> Come to think about it, I do not necessarilly agree that a cluster (not just FreeBSD) with FileStore OSDs in it would warrant a `HEALT_WARN` state. That state is a sort of allert that something in the cluster is requiring more that average attention. And I don't consider having FileStore OSDs in the cluster that high an alert. So if it were up to me, I'd report it in the `ceph status` overview but still mark the cluster HEALTHY

There are performance issues associated with filestore OSDs and a dragging feet for bluestore OSDs (if we have mixture of both objectstore), e.g rgw index pool performance. This warning is in-line to force end user to migrate filestore OSDs to bluestore.

Reply:
> > From my point of view I'd like to remove the warning In HEALTH about depricated FileStore on FreeBSD as long as BlueStore is not available on FreeBSD. Otherwise the cluster will not be able to get into a HEALTHY state. Things like warnings in the logs etc. would be fine.
> > It is just that a FreeBSD cluster with FileStore OSDs is perfectly healty.
> > Come to think about it, I do not necessarilly agree that a cluster (not just FreeBSD) with FileStore OSDs in it would warrant a `HEALT_WARN` state. That state is a sort of allert that something in the cluster is requiring more that average attention. And I don't consider having FileStore OSDs in the cluster that high an alert. So if it were up to me, I'd report it in the `ceph status` overview but still mark the cluster HEALTHY
> 
> There are performance issues associated with filestore OSDs and a dragging feet for bluestore OSDs (if we have mixture of both objectstore), e.g rgw index pool performance. This warning is in-line to force end user to migrate filestore OSDs to bluestore.

If I were to take this to our current cluster, I would get whipped...
It would immediately trigger our monitoring system, with no way to make any serious distinction between this WARN state and more serious warnings. 
No time for the coming weeks/months to do this overhaul,  with the cluster in HEALTH_WARN.
Which is something we normally consider a pre-ERROR state.

It is a preformance issue, nothing is going to break real soon now. So the ignoring sysadmins are allerted, but life can go on. Which is not possible when you set the cluster in HEALTH_WARN
Again, I would leave the cluster healthy because that is exactly what it is, and have a verbose text suggesting upgrading in the sub text like:
```
  cluster:
    id:     0d89e2c3-bcd6-11eb-9014-003048de97cd
    health: HEALTH_OK
        OSD_FILESTORE: 9 osd(s) are on filestore; Filestore has been deprecated, 
              the 'osd_op_queue' is enforced to 'wpq' for filestore OSDs.   

```

Reply:
I think @wjwithagen makes a valid point so if we can get it to work this way, even if this were FreeBSD only behaviour, we should try and accommodate his request.

Reply:
@wjwithagen @badone Ok. @wjwithagen I am good with whatever viable behavior we want to introduce after concluding this discussion. 

How about either of below options ? : 

1.   `ceph health mute OSD_FILESTORE` to get cluster in `HEALTH_OK`, visibly with "(muted: OSD_FILESTORE)" message.

2. Set `mon_warn_on_filestore_osds = false` to not to raise health warn for Filestore OSDs

3. (not so rational) Raise health warn on non-FreeBSD and health ok (with filestore msg) on FreeBSD for filestore OSDs

@neha-ojha @badone

Reply:
What I would like is a combo of 1 and 2.
As long as `mon_warn_on_filestore_osds = false` is not set generate a `HEALTH_WARN` with an explanation.
If sysops set `mon_warn_on_filestore_osds = false`, then the cluster goes to `HEALTH_OK` but still retains the message to keep allerting the operators of the cluster.
And then do not differentiate for FreeBSD.

Reply:
I think the mute option works very well for this, given that it allows users to choose how long they want to mute a particular warning (https://docs.ceph.com/en/latest/rados/operations/monitoring/#muting-health-checks). 

For a deprecation warning like this, I really want us to raise a HEALTH_WARN, especially for non-advanced users, who should know that they need to do something about it. 

@wjwithagen would you agree that the mute option also would work well for your case? then we don't need any special case for FreeBSD in the code.

Reply:
I would perfectly fine with that.
That is why I said, take 1 & 2 in combo.
So if the mute option suppresses `HEALTH_WARN` the that would be fine.

Reply:
@wjwithagen thanks! 

@pdvian I think we are all on the same page now, could you please update the code accordingly, implies 1. no special check for FreeBSD 2. add documentation on how to mute this warning. Thanks for your patience!

Reply:
@neha-ojha Ack. Let me update PR accordingly.